runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MonitoringInfoMetricName.java

```diff
@@ -85,12 +87,33 @@ public String getUrn() {
   }
 
   /** @return The labels associated with this MonitoringInfo. */
-  public HashMap<String, String> getLabels() {
+  public Map<String, String> getLabels() {
     return this.labels;
```

Comment:
Would you mind making this return an unmodifyable map?
        return Collections.unmodifiableMap(map);

This would be safer and prevent adding values to it